scalafmt-core/shared/src/main/scala/org/scalafmt/config/ProjectFiles.scala

```diff
@@ -70,11 +70,26 @@ object ProjectFiles {
     private def regex(seq: Seq[String]) = create(seq, new Regex(_))
 
     private final class Nio(pattern: String) extends PathMatcher {
-      private val matcher = fs.getPathMatcher(pattern)
+      private val matcher =
+        try { fs.getPathMatcher(pattern) }
+        catch {
+          case _: IllegalArgumentException =>
+            throw new ScalafmtConfigException(
+              s"Illegal pattern in configuration: $pattern"
```

Comment:
what happens if we append message from the exception we caught? will it explain the reason better?

Reply:
I tried and it seems empty.

Reply:
Ach, actually for Regex we can add it, it's null for Nio matcher

Reply:
Added `reason: ...` line